jupyterhub/templates/hub/deployment.yaml

```diff
@@ -188,20 +197,23 @@ spec:
             - name: CONFIGPROXY_AUTH_TOKEN
               valueFrom:
                 secretKeyRef:
-                  name: {{ include "jupyterhub.hub-secret.fullname" . }}
-                  key: JupyterHub.proxy_auth_token
+                  # NOTE: References the chart managed k8s Secret even if
+                  #       hub.existingSecret is specified to avoid using the
+                  #       lookup function on the user managed k8s Secret.
+                  name: {{ include "jupyterhub.hub.fullname" . }}
+                  key: ConfigurableHTTPProxy.auth_token
             {{- if .Values.hub.db.password }}
             {{- if eq .Values.hub.db.type "mysql" }}
             - name: MYSQL_PWD
               valueFrom:
                 secretKeyRef:
-                  name: {{ include "jupyterhub.hub-secret.fullname" . }}
+                  name: {{ include "jupyterhub.hub-existing-secret-or-default.fullname" . }}
```

Comment:
What if we set this env in juptyerhub_config instead of the pod? Could we get rid of this exceptional behavior?

Reply:
Absolutely, If you think we can avoid that, it would be great! I has been assuming that we couldn't because it would be consumed by various tools outside our control.

Do you know how we would configure this in a way that covers all situations properly?

Reply:
We can set `os.environ` in jupyterhub_config.py and it should work the same as setting it at the pod level. That will occur soon enough that it will affect the relevant targets (postgres/mysql imports later on in the same process), and also subprocesses, if there are any (there shouldn't be). Only init containers wouldn't get this env, but I don't think that's an issue, and they can always access the same secret if they need to.

So this should work in jupyterhub_config.py (with the appropriate `if`s):

```python
os.environ["MYSQL_PWD"] = values.db.password
```

Reply:
Haha doh oh yes of course!!

Reply:
Resolved!